### PR TITLE
feat(federated-learning): configure private google access

### PR DESCRIPTION
**Pull request by ferrarimarco**
Configure Private Google Access for the federated learning use case

### CODE REVIEW

**Review by arueth — COMMENTED** [review]
This seems overly complex, is there a benefit of doing it all through `for_each` instead of just creating separate `google_dns_managed_zone`s and `google_dns_record_set`s?

File: platforms/gke/base/use-cases/federated-learning/terraform/private_google_access/main.tf [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+locals {
+  # See https://cloud.google.com/vpc/docs/configure-private-google-access#config-domain
+  private_google_access_ips = [
+    "199.36.153.8", "199.36.153.9", "199.36.153.10", "199.36.153.11"
```
[the remark this answers is not in the document]
Comment (ferrarimarco) [reply platforms/gke/base/use-cases/federated-learning/terraform/private_google_access/main.tf]: Done.

**Comment by ferrarimarco** [comment]
> This seems overly complex, is there a benefit of doing it all through `for_each` instead of just creating separate `google_dns_managed_zone`s and `google_dns_record_set`s?

Not the most straightforward approach, but it avoids a lot duplicated resource definitions.

I've added a few comments in order to make this clearer.

Thanks for your review.

**Comment by arueth** [comment]
> Not the most straightforward approach, but it avoids a lot duplicated resource definitions.
> 
> I've added a few comments in order to make this clearer.
> 
> Thanks for your review.

To me, it really obscures what resources are actually being created. Creating complex objects to then create resources makes it difficult to follow. Keeping in mind that we are trying to create easily readable and modifiable code for reference and maintainability.  

If it is the approach you want to take for the use case, I'm fine with it. I think it would be good to get some additional feedback from another team member.

**Comment by ferrarimarco** [comment]
I can refactor this, there will be some duplication tho.

**Comment by ferrarimarco** [comment]
@arueth Done, PTAL. Thanks.

**Comment by arueth** [comment]
Thanks, that is much easier to read and understand. When it comes to IaC, especially for reference, I think using WET is better than DRY. 😀

**Pull request merged**